### ISSUE TITLE
[Help] MDVRP with WorkingTime TimeWindows, but DrivingTime Break

### ISSUE THREAD

**Issue by kd509** [google/or-tools#1652]
Hello,

I started with or-tools some days ago in C# and have a problem now with breaks. Maybe somebody can help me.

I try to solve a routing problem (MDVRP with TimeWindows and Capacity). The TimeWindows are based on working time, which means driving time + service time. So far so good.

What I try to implement now, is a break based on driving time only. So the driver of a vehicle can use the vehicle about 6 hours, but after a maximum of 4.5 hours of driving (!) he needs a break of 45 minutes.

I create the break per vehicle in the following way:

`IntervalVarVector breaks = new IntervalVarVector();`
`IntervalVar breakInterval = solver.MakeFixedDurationIntervalVar(theBreak.timeWindow.Start, theBreak.timeWindow.End, theBreak.Duration, false, $"Break_{b}_{vehicle.Id}");`
`RoutingDimension rdDrivingTime = routing.GetMutableDimension("drivingTime");
breaks.Add(breakInterval);`
`rdDrivingTime.SetBreakIntervalsOfVehicle(breaks, i, node_visit_transits);`

But how to tell the dimension "workingTime", to insert a break at the correct time, because my break bases on drivingTime?

Regards

Karsten

**Comment by Steffaan** [comment]
Hi Karsten,

Did you solve this problem. I'm also looking for a solutions to the EU driver time regulations in .NET?

Regards,

Steffaan

**Comment by kd509** [comment]
> Hi Karsten,
> 
> Did you solve this problem. I'm also looking for a solutions to the EU driver time regulations in .NET?
> 
> Regards,
> 
> Steffaan

No, I guess it is not possible with or-tools until the bug #1730 is not solved.

Regards

Karsten

**Comment by DanielaGuerinoni** [comment]
Hi All, any update on this?

**Comment by metaism-ding** [comment]
Any solution for this MDVRPTW ?
